framework/markdown_test.go

```diff
@@ -0,0 +1,254 @@
+			expected:        "rule-1",
+		},
+		{
+			observation:     ob,
+			observationUuid: "wrong-uuid",
+			expected:        "",
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.observationUuid, func(t *testing.T) {
+			result := extractRuleId(test.observation, test.observationUuid)
+			require.Equal(t, test.expected, result)
+		})
+	}
+}
+
+func TestCreateTemplateValues(t *testing.T) {
+	catalog := oscalTypes.Catalog{
+		Metadata: oscalTypes.Metadata{
+			Title: "Catalog Title",
+		},
+	}
+	assessmentPlan := oscalTypes.AssessmentPlan{
+		LocalDefinitions: &oscalTypes.LocalDefinitions{
+			Components: &[]oscalTypes.SystemComponent{
+				{Title: "Component Title"},
+			},
+		},
+	}
+	assessmentResults := oscalTypes.AssessmentResults{}
+
+	test := struct {
+		catalog           oscalTypes.Catalog
+		assessmentPlan    oscalTypes.AssessmentPlan
+		assessmentResults oscalTypes.AssessmentResults
+		expected          *TemplateValues
+	}{
+		catalog:           catalog,
+		assessmentPlan:    assessmentPlan,
+		assessmentResults: assessmentResults,
+		expected: &TemplateValues{
+			Catalog:           "Catalog Title",
+			Component:         "Component Title",
+			AssessmentResults: assessmentResults,
+		},
+	}
+
+	// Run test
+	result, err := CreateTemplateValues(test.catalog, test.assessmentPlan, test.assessmentResults)
+	if err != nil {
+		t.Errorf("Error creating TemplateValues: %v", err)
+	}
+	require.NoError(t, err)
+	require.Equal(t, test.expected, result)
+}
+
+func TestGenerateAssessmentResultsMd(t *testing.T) {
+	// Mock data for testing
+	assessmentResults := oscalTypes.AssessmentResults{
+		Results: []oscalTypes.Result{
+			{
+				Findings: &[]oscalTypes.Finding{
+					{
+						Target: oscalTypes.FindingTarget{
+							TargetId: "control-1_smt",
+						},
+						RelatedObservations: &[]oscalTypes.RelatedObservation{
+							{
+								ObservationUuid: "observationuuid",
+							},
+						},
+					},
+				},
+				Observations: &[]oscalTypes.Observation{
+					{
+						UUID: "observationuuid",
+						Props: &[]oscalTypes.Property{
+							{
+								Name:  "assessment-rule-id",
+								Value: "rule-value",
+							},
+						},
+						Subjects: &[]oscalTypes.SubjectReference{
+							{
+								SubjectUuid: "subject-1234",
+								Title:       "my component",
+								Props: &[]oscalTypes.Property{
+									{
+										Name:  "result",
+										Value: "fail",
+									},
+									{
+										Name:  "reason",
+										Value: "my reason",
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+
+	templateValues := &TemplateValues{
+		Catalog:           "Catalog Title",
+		Component:         "Component Title",
+		AssessmentResults: assessmentResults,
+	}
+
+	// Run test
+	assessmentResultsMd, err := templateValues.GenerateAssessmentResultsMd("assessment-results.md")
+	if err != nil {
+		t.Errorf("Error generating markdown: %v", err)
+	}
+	expectedmd, err := os.ReadFile("../test/testdata/assessment-results.md")
```

Comment:
It is not critical in this scenario, but I think it could be better to read the file before running the tests. If by any reason the file could not be opened here, the executed tests the previously consumed resources would be useless.

Reply:
Thanks, it sounds reasonable.